pypeit/spectrographs/p200_ngps.py

```diff
@@ -0,0 +1,619 @@
+        retval = super().compound_meta(headarr, meta_key)
+        
+        # If superclass could not handle the meta key
+        if retval is not None:
+            return retval
+        
+        if meta_key == 'mjd':
+            return Time(headarr[0]['UTSHUT']).mjd
+        elif meta_key == 'dispangle':
+            try:
+                return 0
+            except Exception as e:
+                msgs.warn("Could not read dispangle from header:" + msgs.newline() + str(headarr[0]['ANGLE']))
+                raise e    
+            
+        elif meta_key == 'binning':
+            # Always the same binning for DET01 and DET02    
+            binspat = headarr[1]['BINSPAT'] 
+            binspec = headarr[1]['BINSPEC']
+            return f"{binspec},{binspat}"
+        
+        # If there is no target keyword, return image type
+        elif meta_key == 'target':
+            if headarr[0]['TARGET'] is not None:
+                return headarr[0]['TARGET']
+            else:
+                return headarr[0]['IMGTYPE']
+        elif meta_key == 'dichroic': 
+            return None
+        else:
+            msgs.error("Not ready for this compound meta: ", meta_key)
+
+
+    def get_detector_par(self, det: int, hdu: Optional[fits.HDUList] = None):
+        """
+        Return metadata for the selected detector.
+    
+        Args:
+            det (:obj:`int`):
+                1-indexed detector number.
+            hdu (`astropy.io.fits.HDUList`_, optional):
+                The open fits file with the raw image of interest.  If not
+                provided, frame-dependent parameters are set to a default.
+
+        Returns:
+            :class:`~pypeit.images.detector_container.DetectorContainer`:
+            Object with the detector metadata.
+        """
+    
+        binning = self.get_meta_value(self.get_headarr(hdu), 'binning') 
+
+        # Detector 1 (R Channel)
+        detector_dict1 = dict(
+            binning         = binning,
+            det             = 1, # All R channel images assigned to extension 1
+            dataext         = 1, # All R channel images assigned to extension 1
+            specaxis        = 1,
+            specflip        = False, 
+            spatflip        = False, 
+            platescale      = 0.5, 
+            darkcurr        = 0.0,  # e-/pixel/hour (No dark current)
+            saturation      = 65000., # ???
+            nonlinear       = 40./45.,
+            mincounts       = -1e10, # check
+            numamplifiers   = 1, 
+            gain            = np.atleast_1d(2.8),
+            ronoise         = np.atleast_1d(8.5),
+            datasec         = np.atleast_1d(flip_fits_slice(hdu[1].header['DATASEC'])),
+            oscansec        = np.atleast_1d(flip_fits_slice(hdu[1].header['BIASSEC']))
+        )
+
+        # Instantiate
+        detector1 = detector_container.DetectorContainer(**detector_dict1)
+        return detector1
```

Comment:
Skip the middle man: `return detector_container.DetectorContainer(**detector_dict1)`